bci_essentials/eeg_data.py

```diff
@@ -193,8 +193,22 @@ def _pull_data_from_source(self):
         new_eeg_data, new_eeg_timestamps = self.__eeg_source.get_samples()
         new_eeg_data = np.array(new_eeg_data)
 
+        # # Handle when we pass back an empty array
+        # if np.size(new_eeg_data) == 0:
+        #     # Return
+        #     return
+
         # Handle the case when you are using subsets
         if self.__subset != []:
+            # Check to see the dimensions of new_eeg_data bessy will crash if it isn't 2D here.
+            if new_eeg_data.ndim != 2:
+                try:
+                    print(
+                        "Found an issue with the previous data pull. Going to clear the cache and try again"
+                    )
+                except Exception:
```

Comment:
Is this ever going to raise the exception? The preceding try just contains a print statement, so I would think that would work every time.

Reply:
I think you're right.

This PR is going to be cancelled, re: handling the errors in _pull_data_from_source().  This way we deal with invalid data from other sources too.